File: api/src/main/kotlin/es/ubu/lsi/ubumonitorweb/data/api/Module.kt
/*
 * Este fichero forma parte de UBUMonitorWeb.
 *
 * @author Marcelo Verteramo Pérsico
 */

package es.ubu.lsi.ubumonitorweb.data.api

data class Module(
  val id: Int,
  val url: String?,
  val name: String,
  val type: String,
  val plural: String,
  val purpose: String,
  val picture: String,
  val completion: Int,
  val isVisible: Boolean,
  val isUserVisible: Boolean,
  val sinceTs: Long?,
  val untilTs: Long?,
)

File: app/src/app/core/models/module.ts
/*
 * Este fichero forma parte de UBUMonitorWeb.
 *
 * @author Marcelo Verteramo Pérsico
 */

export type Module = {
  id: number;
  url: string | null;
  name: string;
  type: string;
  plural: string;
  picture: string;
  purpose: string;
  completion: number;
  isVisible: boolean;
  isUserVisible: boolean;
  sinceTs: number | null;
  untilTs: number | null;
};
